pkg/internal/checkup/checkup.go

```diff
@@ -183,3 +201,86 @@ func randomizeName(prefix string) string {
 
 	return fmt.Sprintf("%s-%s", prefix, k8srand.String(randomStringLen))
 }
+
+func (c *Checkup) createAndRunTrafficGeneratorPod(ctx context.Context) error {
+	netAttachDef, err := c.client.GetNetworkAttachmentDefinition(
+		ctx,
+		c.namespace,
+		c.params.NetworkAttachmentDefinitionName)
+	if err != nil {
+		return err
+	}
+
+	secondaryNetworksRequest, err := pod.CreateNetworksRequest(map[string]*networkv1.NetworkAttachmentDefinition{
+		c.params.TrafficGeneratorEastMacAddress.String(): netAttachDef,
+		c.params.TrafficGeneratorWestMacAddress.String(): netAttachDef,
+	})
+	if err != nil {
+		return err
+	}
+	trexPod := newTrafficGeneratorPod(c.params, secondaryNetworksRequest)
+
+	if c.trafficGeneratorPod, err = c.client.CreatePod(ctx, c.namespace, trexPod); err != nil {
+		return err
+	}
+
+	if c.trafficGeneratorPod, err = c.waitForPodRunningStatus(ctx, c.namespace, trexPod.Name); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Checkup) waitForPodRunningStatus(ctx context.Context, namespace, name string) (*k8scorev1.Pod, error) {
+	log.Printf("Waiting for Pod %s/%s..", namespace, name)
+	var updatedPod *k8scorev1.Pod
+
+	conditionFn := func(ctx context.Context) (bool, error) {
+		var err error
+		updatedPod, err = c.client.GetPod(ctx, namespace, name)
+		if err != nil {
+			return false, err
+		}
+		return pod.PodInRunningPhase(updatedPod), nil
+	}
+	const interval = time.Second * 5
+	if err := wait.PollImmediateUntilWithContext(ctx, interval, conditionFn); err != nil {
+		return nil, fmt.Errorf("failed to wait for POD '%s/%s' to be in Running Phase: %v", namespace, name, err)
```

Comment:
Please consider using the ObjectFullName function.

Reply:
DONE